kernel/tests/read.rs

```diff
@@ -706,8 +694,143 @@ fn mixed_not_null() -> Result<(), Box<dyn std::error::Error>> {
     read_table_data_str(
         "./tests/data/mixed-nulls",
         Some(&["part", "n"]),
-        Some(Expression::column("n").is_null().not()),
+        Some(Expression::not(Expression::column("n").is_null())),
         expected,
     )?;
     Ok(())
 }
+
+#[test]
+fn and_or_predicates() -> Result<(), Box<dyn std::error::Error>> {
+    let cases = vec![
+        (
+            Expression::column("number")
+                .gt(Expression::literal(4i64))
+                .and(Expression::column("a_float").gt(Expression::literal(5.5))),
```

Comment:
nit: AND/OR are similar to NOT, in being hard(er) to read in this "chained" form.
```suggestion
            Expression::and(
                Expression::column("number").gt(Expression::literal(4i64)),
                Expression::column("a_float").gt(Expression::literal(5.5)),
            ),
```
It does admittedly add one extra line to the expression tho. Your call which is better overall.

Reply:
Also, just double checking: `a_float` is a f64 column (not f32), right?